### PR TITLE
[feature-1] Added Binary Search in Kotlin

### LINKED ISSUE #2312: Iterative Binary Search in Kotlin
## 🚀 Feature

Binary Search is a better searching algorithm than linear search in a sorted collection of elements. I would like to implement the iterative version of it in Kotlin.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
YES

## Motivation
There are no searching algorithms in the Kotlin directory of this repository, so it just motivated me to add one.

## Pitch
Binary Search algorithm is the most used searching algorithm. Also, it has various uses like finding the first and last occurrence of a key, finding the peak element, etc. Adding this algorithm would help to solve other problems related to it.

### PR DESCRIPTION
**Pull request by kiruba-r11**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes

### Description
Added Iterative Binary Search Algorithm in Kotlin.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Closes: #2312

### CODE REVIEW

**Comment by kiruba-r11** [comment]
@HarshCasper and @bhav09 Please review this.

File: Kotlin/search/BinarySearch.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Test - 2:
+    Input
+        5
+        10 12 45 67 89
+        1000
+
+    Output
+        Key: 1000 is not present in the array
+ */
```
Comment (bhav09) [inline Kotlin/search/BinarySearch.kt]: Add a new line here

**Review by bhav09 — CHANGES_REQUESTED** [review]
It's perfect! However, you can add a new line at the end of the code.

**Comment by kiruba-r11** [comment]
@bhav09 I have added it👍

**Comment by bhav09** [comment]
Great!

**Comment by kiruba-r11** [comment]
@HarshCasper @bhav09 Please add the level and gssoc label

**Pull request merged**